File: 0649-dota2-senate/0649-dota2-senate.java
class Solution {
    public String predictPartyVictory(String senate) {
        Queue<Integer> Radiant = new LinkedList<>();
        Queue<Integer> Dire = new LinkedList<>();
        int n = senate.length();

        for(int i = 0; i<n; i++){
            if(senate.charAt(i) == 'R'){
                Radiant.offer(i);
            }else{
                Dire.offer(i);
            }
        }

        while(!Radiant.isEmpty() && !Dire.isEmpty()){
            int radiant = Radiant.poll();
            int dire = Dire.poll();
            n++;
            if(radiant < dire){
                Radiant.offer(n);
            }else{
                Dire.offer(n);
            }
        }
        String answer = (!Radiant.isEmpty())? "Radiant" : "Dire";
        return answer;
    }
}

File: 0649-dota2-senate/0649-dota2-senate.py
class Solution:
    def predictPartyVictory(self, senate: str) -> str:
        radiantQueue = []
        direQueue = []
        size = len(senate)

        for i in range(size):
            if senate[i] == 'R':
                radiantQueue.append(i)
            else:
                direQueue.append(i)
        
        while radiantQueue and direQueue:
            r,q = radiantQueue.pop(0), direQueue.pop(0)

            if r < q:
                radiantQueue.append(r + size)
            else:
                direQueue.append(q + size)

        return "Dire" if len(direQueue) > 0 else "Radiant"